src/frontend/js/components/Icon/index.tsx

```diff
@@ -0,0 +1,58 @@
+interface IconProps {
+  name: IconType;
+  /**
+   * Optional title, will be announced by screen readers
+   * and set as `title` attribute for mouse users
+   *
+   * If omitted, the icon is considered decorative and
+   * ignored by screen readers
+   */
+  title?: string;
+  className?: string;
+}
+
+export const Icon = ({ name, title, className = '' }: IconProps) => {
+  return (
+    <svg
+      {...{
+        className: `icon ${className}`,
+        ...(title
+          ? {
+              role: 'img',
+              'aria-label': title,
+              title,
+            }
+          : {
+              'aria-hidden': true,
+            }),
+      }}
+    >
```

Comment:
What do you think to rewrite this like that in order to improve readability?
```suggestion
    <svg
      className={`icon ${className}`}
      aria-hidden={!title}
      {...(title && {
        role: 'img',
        'aria-label': title,
        title,
      })}
    >
```

Reply:
Yeah, it's better!